test/sink.test.js

```diff
@@ -149,5 +149,6 @@ test('should destroy writable stream if source throws', async t => {
   const err = await t.throwsAsync(pipe(source, toIterable.sink(stream)))
 
   t.is(err.message, 'boom')
+  t.false(stream.writable)
```

Comment:
This test passes on master

Reply:
I've added a bunch of assertions to make it clear when we expect the stream to be made non-writable and when we expect it to be destroyed, so some of those will pass already.

I've also added more tests that try to hit some of the edge cases and also added tests for slow iterators because we were finishing our `source` in the tests before error events were being emitted that otherwise lead to crashes.

Reply:
> I've added a bunch of assertions to make it clear when we expect the stream to be made non-writable and when we expect it to be destroyed

You know, I say that, and then node 14 goes and changes `.destroyed` to true all on its own when the `close` event is emitted.  Uff.

Reply:
Turns out node 14 flips the `autoDestroy` option default value to true.  I've specified it as false in the tests so we can assert on whether we've destroyed the stream or not.